`ts: "esbuild"` crashes

`ts: "esbuild"` crashes with error message `Cannot create property 'sourcesContent' on string`.

<details><summary>Full error message</summary>

```
✘ [ERROR] Cannot create property 'sourcesContent' on string '{
  "version": 3,
  "sources": ["src\\main.civet.jsx"],
  "sourcesContent": ["import {a} from \"./module.civet\"\n\nconsole.log(a)\n"],
  "mappings": "AAAA,SAAQ,SAAQ;AAEhB,QAAQ,IAAI,CAAC;",
  "names": []
}
' [plugin unplugin-civet]

  This error came from the "onLoad" callback registered here:

    node_modules/esbuild/lib/main.js:1293:20:
      1293 │       let promise = setup({
           ╵                     ^

    at file:///C:/Users/edemaine/Projects/Civet/node_modules/unplugin/dist/index.mjs:1173:9
    at handlePlugins (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:1293:21)
    at buildOrContextImpl (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:979:5)
    at Object.buildOrContext (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:787:5)
    at C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:2188:15
    at new Promise (<anonymous>)
    at Object.build (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:2187:25)
    at Module.build (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:2036:51)
    at file:///C:/Users/edemaine/Projects/Civet/integration/unplugin/examples/esbuild/esbuild.js:17:17

C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:1650
  let error = new Error(text);
              ^

Error: Build failed with 1 error:
error: Cannot create property 'sourcesContent' on string '{
  "version": 3,
  "sources": ["src\\main.civet.jsx"],
  "sourcesContent": ["import {a} from \"./module.civet\"\n\nconsole.log(a)\n"],
  "mappings": "AAAA,SAAQ,SAAQ;AAEhB,QAAQ,IAAI,CAAC;",
  "names": []
}
'
    at failureErrorWithLog (C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:1650:15)
    at C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:1059:25
    at C:\Users\edemaine\Projects\Civet\integration\unplugin\examples\esbuild\node_modules\esbuild\lib\main.js:1526:9
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  errors: [Getter/Setter],
  warnings: [Getter/Setter]
}
```
</details>

**Reproduction:** modify our `integration/unplugin/examples/esbuild` example to use `ts: 'esbuild'` as follows:

```diff
diff --git a/integration/unplugin/examples/esbuild/esbuild.js b/integration/unplugin/examples/esbuild/esbuild.js
index ae35042..f76625a 100644
--- a/integration/unplugin/examples/esbuild/esbuild.js
+++ b/integration/unplugin/examples/esbuild/esbuild.js
@@ -5,7 +5,9 @@ const options = {
   entryPoints: ['src/main.civet'],
   bundle: true,
   outfile: 'dist/main.js',
-  plugins: [civetEsbuildPlugin()],
+  plugins: [civetEsbuildPlugin({
+    ts: 'esbuild',
+  })],
 }

 if (process.argv.includes('--watch')) {
```

--- a/integration/unplugin/examples/esbuild/esbuild.js
+++ b/integration/unplugin/examples/esbuild/esbuild.js
@@ -5,7 +5,9 @@ const options = {
   entryPoints: ['src/main.civet'],
   bundle: true,
   outfile: 'dist/main.js',
-  plugins: [civetEsbuildPlugin()],
+  plugins: [civetEsbuildPlugin({
+    ts: 'esbuild',
+  })],
 }
 
 if (process.argv.includes('--watch')) {

--- a/integration/unplugin/src/index.ts
+++ b/integration/unplugin/src/index.ts
@@ -318,7 +318,7 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
 
       const jsonSourceMap =
         typeof compiled.sourceMap == 'string'
-          ? compiled.sourceMap
+          ? JSON.parse(compiled.sourceMap)
           : compiled.sourceMap.json(
               path.basename(id.replace(/\.[jt]sx$/, '')),
               path.basename(id)
@@ -326,7 +326,7 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
 
       let transformed: TransformResult = {
         code: compiled.code,
-        map: JSON.parse(jsonSourceMap as any),
+        map: jsonSourceMap,
       };
 
       if (options.transformOutput)

--- a/integration/unplugin/src/index.ts
+++ b/integration/unplugin/src/index.ts
@@ -1,8 +1,4 @@
-import {
-  TransformResult,
-  createUnplugin,
-  SourceMapCompact as UnpluginSourceMap,
-} from 'unplugin';
+import { TransformResult, createUnplugin } from 'unplugin';
 import civet, { SourceMap } from '@danielx/civet';
 import {
   remapRange,
@@ -330,7 +326,7 @@ const civetUnplugin = createUnplugin((options: PluginOptions = {}) => {
 
       let transformed: TransformResult = {
         code: compiled.code,
-        map: jsonSourceMap as UnpluginSourceMap,
+        map: JSON.parse(jsonSourceMap as any),
       };
 
       if (options.transformOutput)

